### PR TITLE
[Misc] main2main upgrade vllm to 0619

### PR DESCRIPTION
**Pull request by SOMEONEUNSEEN**
### What this PR does / why we need it?
This PR adapts vllm-ascend to the latest upstream vLLM changes (commit `560fb8b867aaa444d471b35fd846368ebacf12b9`). The changes are organized by the upstream PRs they correspond to:

#### 1. Dynamic Speculative Decoding Support (#32374)

**Files changed:**

- `vllm_ascend/worker/model_runner_v1.py`

- `vllm_ascend/spec_decode/suffix_proposer.py`

- `vllm_ascend/worker/v2/model_runner.py`

**Changes:**

- Updated `propose()` method signatures for `AscendNgramProposer`, `AscendSuffixDecodingProposer`, `AscendMedusaProposer`, and `AscendExtractHiddenStatesProposer` to accept `num_speculative_tokens` as the first parameter

- Added `import torch` to `suffix_proposer.py` to fix undefined name error

- Used temporary value for `decode_query_len` initialization in v2 model runner

#### 2. Model Runner V2 Migration (#44446)

**Files changed:**

- `vllm_ascend/patch/platform/patch_pp_mtp.py`

**Changes:**

- Added `_patch_default_v2_model_runner_architectures()` to remove `DeepseekV2ForCausalLM` from `DEFAULT_V2_MODEL_RUNNER_ARCHITECTURES`

- This ensures DeepSeek-V2 models use the v1 model runner which has proper PP (pipeline parallel) support for MoE models

#### 3. Streaming Parser Engine (#45915)

**Files changed:**

- `vllm_ascend/patch/platform/patch_glm47_tool_call_parser.py`

- `vllm_ascend/patch/platform/patch_minimax_m2_tool_call_parser.py`

- `tests/ut/patch/platform/test_patch_glm47_tool_call_parser.py`

- `tests/ut/patch/platform/test_patch_minimax_m2_tool_call_parser.py`

**Changes:**

- Wrapped patches with `if False:` since upstream has replaced old regex-based parsers with the new state-machine engine

- Set `_LEGACY_PARSER = False` in tests to skip legacy parser test cases

#### 4. Anthropic System Message Support (#44602, #46025)

**Files changed:**

- `vllm_ascend/patch/platform/patch_anthropic_system_message.py`

- `tests/ut/patch/platform/test_patch_anthropic_system_message.py`

**Changes:**

- Wrapped patches with `if False:` since upstream now natively supports `merge_inline_system` parameter

- Added `_LEGACY_MODE` detection in tests to skip when upstream has the feature

- Updated tests to pass `merge_inline_system=True` parameter

#### 5. Scheduler Throttle Prefills Parameter

**Files changed:**

- `vllm_ascend/patch/platform/patch_balance_schedule.py`

**Changes:**

- Added `throttle_prefills: bool = False` parameter to `schedule()` method

- Forwarded parameter to parent class method

#### 6. Regex Compilation Timeout Fix (#45118)

**Files changed:**

- `tests/e2e/pull_request/one_card/test_guided_decoding.py`

**Changes:**

- Set `VLLM_REGEX_COMPILATION_TIMEOUT_S=30` for `xgrammar` and `outlines` backends

- This handles complex JSON schemas that may exceed the default 5s timeout

#### 7. Test Infrastructure Improvements

**Files changed:**

- `tests/ut/patch/platform/test_patch_minimax_usage_accounting.py`

**Changes:**

- Added missing `eos_token_id`, `bos_token_id`, `pad_token_id` attributes and `decode()` method to `FakeTokenizer` class for compatibility with new state-machine engine

### Does this PR introduce any user-facing change?

**No.**  All changes are internal adaptations to upstream vLLM API changes. Users should not observe any behavioral differences after this update.

**Key points:**

- All patches are backward-compatible with older upstream versions via conditional checks

- DeepSeek-V2 models will continue to work correctly with PP configurations

- No changes to public APIs or configuration options

### Related Upstream PRs

| Upstream PR | Description | Files Affected |
|---|---|---|
| #32374 | [V1][Spec Decode] Add Dynamic SD | model_runner_v1.py, suffix_proposer.py, model_runner.py (v2) |
| #44446 | [Model Runner V2] Migration to support quantized model by default [5/N] | patch_pp_mtp.py |
| #45915 | [Frontend] Add Streaming Parser Engine and new GLM4.7/GLM5.1/GLM5.2 Parser | patch_glm47_tool_call_parser.py, patch_minimax_m2_tool_call_parser.py |
| #44602 | fix(anthropic): preserve inline system message position for prefix caching | patch_anthropic_system_message.py |
| #46025 | fix(anthropic): auto-detect template support for mid-conversation system messages | patch_anthropic_system_message.py |
| #45118 | [Security] Add timeout guard for regex compilation in structured output | test_guided_decoding.py |

### How was this patch tested?
https://github.com/vllm-project/vllm-ascend/actions/runs/27892595280/job/82538857415?pr=10729
https://github.com/vllm-project/vllm-ascend/actions/runs/27896389470/job/82548948104?pr=10729
https://github.com/vllm-project/vllm-ascend/actions/runs/27899594634/job/82557641743?pr=10729
https://github.com/vllm-project/vllm-ascend/actions/runs/27903089171/job/82567241257?pr=10729
https://github.com/vllm-project/vllm-ascend/actions/runs/27928484787/job/82635950403?pr=10729
https://github.com/vllm-project/vllm-ascend/actions/runs/27935695692/job/82657424062?pr=10729
https://github.com/vllm-project/vllm-ascend/actions/runs/27935695692/job/82657424065?pr=10729

- vLLM version: v0.23.0
- vLLM main: https://github.com/vllm-project/vllm/commit/967c5c3bc38891f4465d3f4e99917ed837bb3833

### CODE REVIEW

File: .github/workflows/nightly_image_build.yaml [hunk 1]
```diff
@@ -26,10 +26,11 @@ on:
       vllm_ascend_branch:
         description: 'Branch to nightly test'
         required: true
-        default: 'main'
```
Comment (wangxiyuan) [inline .github/workflows/nightly_image_build.yaml]: 0619 has contained 0.23.0, why not change release tag to 0.23.0 thin this PR as the same time?
